src/figuresClasses.ts

```diff
@@ -1,11 +1,86 @@
-export interface Figure {}
+type ValidShapes = 'triangle' | 'circle' | 'rectangle';
+type ValidColors = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: ValidShapes;
+  color: ValidColors;
+  getArea: () => number;
+}
+
+export class Triangle implements Figure {
+  public shape: ValidShapes = 'triangle';
+
+  constructor(
+    public color: ValidColors,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sides = [a, b, c];
+    const longestSide = Math.max(...sides);
+    let sumOfOtherSides = 0;
+
+    for (const side of sides) {
+      if (side < longestSide) {
+        sumOfOtherSides += side;
+      }
+    }
+
+    if (a <= 0 || b <= 0 || c <= 0 || sumOfOtherSides <= longestSide) {
+      throw new Error('is not valid length!');
```

Comment:
[CODE STYLE] - If you throw an error try to provide its detailed description in error message. Instead of a generic 'is not valid length!', provide a message that explains the specific issue, like 'Triangle sides must be positive numbers and the sum of any two sides must be greater than the third side.'